### PR TITLE
fix: fix helm chart 4.4.x errors

### PR DESCRIPTION
**Pull request by mohammadiahmad**
unclosed action in StatefulSet.yaml
comparison error in configmap.env.yaml

### CODE REVIEW

File: deploy/charts/emqx/templates/StatefulSet.yaml [hunk 1]
```diff
@@ -1,8 +1,4 @@
-{{ $configData := printf "%s\n%s\n%s\n%s"
-  (include (print $.Template.BasePath "/configmap.env.yaml") .)
-  (include (print $.Template.BasePath "/configmap.acl.yaml") .)
-  (include (print $.Template.BasePath "/configmap.loadedPlugins.yaml") .)
-  (include (print $.Template.BasePath "/configmap.loadedModules.yaml") .) }}
+{{ $configData := printf "%s\n%s\n%s\n%s"  (include (print $.Template.BasePath "/configmap.env.yaml") .)  (include (print $.Template.BasePath "/configmap.acl.yaml") .)   (include (print $.Template.BasePath "/configmap.loadedPlugins.yaml") .)   (include (print $.Template.BasePath "/configmap.loadedModules.yaml") .) }}
```
Comment (zmstone) [inline deploy/charts/emqx/templates/StatefulSet.yaml:1]: what are the alternatives to make it multi-line so it is more readable ?

maybe assign 5 vars so the printf args list can be shorter ? 

or, maybe use multi-line values `|-`
this `$configData` is only used to generate a checksum, so the format isn't important.

Comment (mohammadiahmad) [reply deploy/charts/emqx/templates/StatefulSet.yaml:1]: We cannot split the ``printf`` Go template function into multiple line.

[https://stackoverflow.com/questions/49816911/how-to-split-a-long-go-template-function-across-multiple-lines](url)

File: deploy/charts/emqx/templates/configmap.env.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ metadata:
     app.kubernetes.io/managed-by: {{ .Release.Service }}
 data:
   {{- range $index, $value := .Values.emqxConfig }}
-  {{- if ne $value nil }}
+  {{- if $value }}
```
[the remark this answers is not in the document]
Comment (zmstone) [reply deploy/charts/emqx/templates/configmap.env.yaml:14]: sorry, it was fixed in `main-v4.3`, but not `main-v4.4`.

File: deploy/charts/emqx/templates/StatefulSet.yaml (hunk 1 above)
Comment (zmstone) [reply deploy/charts/emqx/templates/StatefulSet.yaml:1]: let's do that.

Comment (mohammadiahmad) [reply deploy/charts/emqx/templates/StatefulSet.yaml]: I refactored the `$configData` using multiple variable.

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
Hi @mohammadiahmad 
as it turns out it's broken in 4.3 too, 
so I have squashed your commits and rebased on to `main-v4.3`
https://github.com/emqx/emqx/pull/7218

thank you so much for the PR.

**Comment by zmstone** [comment]
just to clarify, `main-v4.3` is periodically forward merged to `main-v4.4`,
so there is no need to PR separately.

**Pull request closed without merging**